### PR TITLE
Make sure rid specific libs projs get packaged

### PR DESCRIPTION
**Pull request by ViktorHofer**
**Make sure rid specific libs projs get packaged …**

RID specific runtime.*.runtime.native.System.IO.Ports projects weren't
packaged because the GeneratePackageOnBuild property wasn't set to true
for them. The property was only true during an allconfigurations build.
As these projects are only built outside of an allconfigurations build,
the GeneratePackageOnBuild property needs to account for such as well.

Also updating the NoTargets Sdk to clean the rid specific package up.

Moving the GeneratePackageOnBuild logic into a props file and set it to
false during servicing (or runtimelab) so that devs can set the property
during servicing directly in the project file if a project should be
packaged.

**Remove the GeneratePackage property …**

That property was intended to enabled incremental servicing but with
now using the NuGet Pack task we can just make use of the publicly
available GeneratePackageOnBuild property.

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -54,24 +54,8 @@
     <PackageDescription Condition="'$(PackageDescription)' == '' and '$(UseRuntimePackageDisclaimer)' == 'true'">$(RuntimePackageDisclaimer)</PackageDescription>
     <!-- Keep in sync as required by the Packaging SDK in Arcade. -->
     <Description>$(PackageDescription)</Description>
-    <GenerateNuspecDependsOn>ErrorForMissingPackageDescription;$(GenerateNuspecDependsOn)</GenerateNuspecDependsOn>
   </PropertyGroup>
 
-  <!-- Remove when https://github.com/NuGet/Home/issues/10405 is implemented and consumed. -->
-  <Target Name="ErrorForMissingPackageDescription"
```
Comment (ViktorHofer) [inline Directory.Build.targets:61]: Also removing this block as it didn't work anymore for the NuGet pack as PackageDescription is always set to a default value.

**Review by hoyosjs — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Formatting job failure: https://github.com/dotnet/runtime/issues/57196

**Comment by ViktorHofer** [comment]
runtime (Libraries Test Run release coreclr windows x86 Debug) failure is https://github.com/dotnet/runtime/issues/57187

**Comment by ViktorHofer** [comment]
runtime (Libraries Test Run checked coreclr Linux_musl arm64 Release) failure is https://github.com/dotnet/runtime/issues/57198
The other two failing legs are ttps://github.com/dotnet/runtime/issues/55449

Merging as the previous run already completed successfully and only the packaging inside the Libraries Linux x64 leg changed which I just verified is correct now.

**Pull request merged**